tests/components/FilterSearch.test.tsx

```diff
@@ -104,74 +104,90 @@ describe('search with section labels', () => {
+
+      userEvent.type(searchBarElement, 'n');
+      const autocompleteSuggestion = await waitFor(() => screen.findByText('first name 1'));
+
+      const expectedSetFilterOptionParam = {
+        fieldId: 'name',
+        matcher: Matcher.Equals,
+        value: 'first name 1',
+        displayName: 'first name 1',
+        selected: true
+      };
+      const expectedSetOffsetParam = 0;
+
+      userEvent.click(autocompleteSuggestion);
+      expect(setFilterOption).toBeCalledWith(expectedSetFilterOptionParam);
+      expect(setOffset).toBeCalledWith(expectedSetOffsetParam);
+
+      const setFilterOptionCallOrder = setFilterOption.mock.invocationCallOrder[0];
+      const setOffsetCallOrder = setOffset.mock.invocationCallOrder[0];
+      const mockExecuteSearchCallOrder = mockExecuteSearch.mock.invocationCallOrder[0];
+      expect(setFilterOptionCallOrder).toBeLessThan(mockExecuteSearchCallOrder);
+      expect(setOffsetCallOrder).toBeLessThan(mockExecuteSearchCallOrder);
     });
-    expect(setOffset).not.toBeCalled();
-    expect(mockExecuteSearch).not.toBeCalled();
   });
 
-  it('triggers a search when an autocomplete result is clicked', async () => {
-    const mockExecuteSearch = jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
-    render (<FilterSearch searchFields={searchFieldsProp}/>);
-    const searchBarElement = screen.getByRole('textbox');
+  describe('searchOnSelect = false', () => {
+    it('does not trigger a search when an autocomplete result is selected', async () => {
+      const mockExecuteSearch = jest.spyOn(searchOperations, 'executeSearch').mockImplementation();
+      render(<FilterSearch searchFields={searchFieldsProp} searchOnSelect={false} />);
+      const searchBarElement = screen.getByRole('textbox');
 
-    userEvent.type(searchBarElement, 'n');
-    const autocompleteSuggestion = await waitFor(() => screen.findByText('first name 1'));
-
-    const expectedSetFilterOptionParam = {
-      fieldId: 'name',
-      matcher: Matcher.Equals,
-      value: 'first name 1' ,
-      displayName: 'first name 1',
-      selected: true
-    };
-    const expectedSetOffsetParam = 0;
-
-    userEvent.click(autocompleteSuggestion);
-    expect(setFilterOption).toBeCalledWith(expectedSetFilterOptionParam);
-    expect(setOffset).toBeCalledWith(expectedSetOffsetParam);
-
-    const setFilterOptionCallOrder = setFilterOption.mock.invocationCallOrder[0];
-    const setOffsetCallOrder = setOffset.mock.invocationCallOrder[0];
-    const mockExecuteSearchCallOrder = mockExecuteSearch.mock.invocationCallOrder[0];
-    expect(setFilterOptionCallOrder).toBeLessThan(mockExecuteSearchCallOrder);
-    expect(setOffsetCallOrder).toBeLessThan(mockExecuteSearchCallOrder);
+      userEvent.type(searchBarElement, 'n');
+      await waitFor(() => screen.findByText('first name 1'));
+
+      userEvent.type(searchBarElement, '{arrowdown}{enter}');
+      expect(mockExecuteSearch).not.toHaveBeenCalled();
```

Comment:
can we add a check that `setFilterOption` is still called?

Reply:
updated!